apps/web-giddh/src/app/ledger/ledger.component.ts

```
@@ -3659,20 +3653,20 @@ export class LedgerComponent implements OnInit, OnDestroy {
                         variantDiscount = stock.variant?.variantDiscount;
                         quantity = txn?.inventory?.quantity || 1;
                     } else {
-                        const defaultUnitRates = this.generalService.voucherApiVersion === 1 ? txn.selectedAccount?.stock?.unitRates : txn.selectedAccount?.stock?.variant?.unitRates;
+                        stockUnitUniqueName = txn.selectedAccount.stock.stockUnitUniqueName;
+                        const defaultUnitRates = (this.generalService.voucherApiVersion === 1 ? txn.selectedAccount?.stock?.unitRates : txn.selectedAccount?.stock?.variant?.unitRates).filter(unitRate => unitRate.stockUnitUniqueName === stockUnitUniqueName);
```

Biome (2.3.13)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)